src/bin/stats.rs

```diff
@@ -4,16 +4,17 @@ use std::fs::File;
 use std::env;
 use std::io::prelude::*;
 
-use fuzzy_pickles::{Visit, Visitor};
+use fuzzy_pickles::{Control, Visit, Visitor};
 
 #[derive(Debug, Default)]
 struct Stats {
     statements: usize,
 }
 
 impl Visitor for Stats {
-    fn visit_statement(&mut self, _: &fuzzy_pickles::Statement) {
+    fn visit_statement(&mut self, _: &fuzzy_pickles::Statement) -> Control {
```

Comment:
What happened to your idea of returning `impl Into<Control>`? While that would require changes on this line, it would save the change at the end of the method, right?

Reply:
To be overly dramatic: reality happened!

So according to the irc, current impl trait doesn't allow polymorphism / a certain type has to be returned. Because there can be multiple implementations for a trait (on different types oc), trait functions shall not return impl traits because they could have different types in their respective impl

Reply:
In my example, sadly, i did only prove it working for functions. If i would've tried it with traits, we would've hit this earlier on.